src/object/thunderstorm.cpp

```diff
@@ -94,19 +96,35 @@ Thunderstorm::update(float )
     lightning();
     time_to_thunder.start(interval);
   }
+
+  if(flash_display_timer.started()) {
+    float alpha = 0.9f;
+    if(flash_display_timer.get_timegone() > 0.1f) {
+      auto progress = flash_display_timer.get_timegone() / flash_display_timer.get_timeleft() - 0.1f;
+      if(progress < 0.0f)
+        progress = 0.0f;
+
+      alpha = 0.9f - progress;
+    }
+    
+    if(alpha < 0.0f) {
+      flash_display_timer.stop();
+      return;
+    }
+
+    m_flash_color = Color(alpha, alpha, alpha, 1.0);
+  }
 }
 
 void
 Thunderstorm::draw(DrawingContext& context)
 {
   if (!flash_display_timer.started()) return;
 
-  float alpha = 0.33f;
   context.push_transform();
   context.set_translation(Vector(0, 0));
-  context.color().draw_filled_rect(context.get_rect(), Color(1, 1, 1, alpha), layer);
+  context.color().draw_gradient(m_flash_color, m_flash_color, 500, GradientDirection::HORIZONTAL, context.get_rect(), Blend::ADD);
```

Comment:
Before the gradient is drawn, the `DrawingContext` transform scale should be set to 1:

```
context.transform().scale = 1.f;
```

Doing so will prevent #2770.